ui/src/childinfo/ChildInfo.js

```diff
@@ -1,177 +1,204 @@
 
 import React, { Component } from 'react';
+import WishDetailsService from '../services/WishDetailsService';
+import WishDetails from '../wishDetails/WishDetails';
 import rocketImage from '../../src/assets/icn_To_Go_Rocket_White_Inside_130x130.png';
 import './ChildInfo.css';
 
 export default class ChildInfo extends Component {
-  constructor(props){
-      super(props);
-      this.props = props;
-      this.state = {
-        step: 0,
-        homeTown: '',
-        illness: '',
-        details: '',
-        showConfirmation: false,
-        rocketRotation: 20,
-        rocketWidth: 170,
-        rocketContainerHeight: 350,
-        isBlastOff: false
+  constructor(props) {
+    super(props);
+    this.props = props;
+    this.state = {
+      step: 0,
+      homeTown: '',
+      illness: '',
+      details: '',
+      showConfirmation: false,
+      showWishDetails: false,
+      rocketRotation: 20,
+      rocketWidth: 170,
+      rocketContainerHeight: 350,
+      isBlastOff: false
+    }
+    this.numSteps = Object.keys(this.stepMapFunction()).length
+  }
+
+  stepMapFunction = () => {
+    let { name } = this.props
+    return {
+      0: {
+        text: `Hi${name ? ` ${name}` : ''}, I have a few questions for you before we can make your wish come true!`,
+        input: ''
+      },
+      1: {
+        text: 'Where are you from?',
+        input: 'homeTown'
+      },
+      2: {
+        text: 'Tell us about your condition',
+        input: 'illness'
+      },
+      3: {
+        text: 'Tell us more about your wish!',
+        input: 'details'
       }
-      this.numSteps = Object.keys(this.stepMapFunction()).length
     }
+  }
 
-    stepMapFunction = () => {
-      let { name } = this.props
-      return {
-        0: {
-          text: `Hi${name ? ` ${name}` : ''}, I have a few questions for you before we can make your wish come true!`,
-          input: ''
-        },
-        1: {
-          text: 'Where are you from?',
-          input: 'homeTown'
-        },
-        2: {
-          text: 'Tell us about your condition',
-          input: 'illness'
+  nextStep = () => {
+    let stepMap = this.stepMapFunction();
+    let { step } = this.state;
+    if (step < Object.keys(stepMap).length - 1) {
+      window.scrollTo({
+        top: 1000,
+        behavior: 'smooth'
+      });
+      this.setState({
+        step: step + 1
+      }, () => {
+        this.scrollToTop();
+      });
+    } else {
+
+      const wish = {
+        child: {
+          name: this.state.name,
```

Comment:
We should destructure all these variables to be consistent with rest of the project